base/osutils.jl

```diff
@@ -1,20 +1,26 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
-"""
-    OS_NAME
+let UNAME = ccall(:jl_get_UNAME, Any, ())
+    # change some `uname -s` values to "friendly" names
+    # for internal usage
```

Comment:
> cygwin isn't a kernel, it's a libc.

I realize that it also implements a libc, but it also implements a kernel (pipes, tty, fork, exec, etc. are provided by cygwin, not nt). Given that it implements a kernel, what would you call it?

> What do you mean by hybrid applications?

Every posix implementation has unique features that we make use of to create a hybrid between the pure posix definition and the "native" API for that kernel. The most obvious place where this is visible is for GUI toolkits, but occasionally comes up in other places like signal handling, event descriptor, and messages vs. proc.

> libuv doesn't call into the kernel on windows, it calls the win32 api

This is an implementation flaw, not a feature. Unfortunately, Microsoft doesn't document NT directly, and only (currently) provides access to the kernel by exposing its syscalls as Win32 functions. This leads to confusion between whether a particular function is a library or kernel call.

> midipix is also userspace only

midipix would allow us to bypass the restrictions of the win32 wrapper and access the `nt` kernel directly. the permission ring of the processor isn't an essential part of the definition of a kernel. but indeed, win32 and midipix do both provide complementary access to the nt kernel.

> They released WSL, and we won't be getting anything dramatically different

unfortunately true, as I said above: "Initial reports seem indicate that Microsoft's kernel is going to lie and call itself Linux. This is highly unfortunate..."

> WSL largely makes midipix obsolete.

not currently, and unless midipix dies or fails, I don't see this ever happening. midipix is what WSL should have been: a native API that supplemented Win32. If you look at the website for midipix (http://midipix.org) it is intended to offer a strict superset of what WSL appears capable of providing.